developer/services/disposablevm-implementation.rst

```diff
@@ -2,73 +2,162 @@
 
-3. create the ``/var/run/qubes/fast_block_attach`` file, whose presence tells the ``/etc/xen/scripts/block`` script to bypass some redundant checks and execute as fast as possible.
+- Order this action after the autostart or standard qubes;
+- Skip preloading if kernel command line prevents autostart.
 
-4. execute ``xl restore`` in order to restore a domain.
+Preloaded disposable's memory management
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-5. create the same xenstore keys as normally created when AppVM boots (e.g. ``qubes_ip``)
 
-6. create the ``qubes_restore_complete`` xenstore key. This allows the boot process in DisposableVM to continue.
+At the end of preloading, the qube is paused if it has not been requested yet, but before pausing, on :py:meth:`domain-pre-paused <core-admin:qubes.vm.dispvm.DispVM.on_domain_pre_paused>`, it attempts to retrieve memory from the qube by setting it to use its preferred memory value, which in :doc:`qmemman terms </developer/services/qmemman>`, is just enough to have the qube running. The memory must be managed before the qube is paused, cause once on the paused state, it is not possible to negotiate with the domain.
 
+This process can take a bit of time because it depends on how fast the qube can free up memory. There is a timeout and a threshold in transfer speed. When any of these exit conditions are met, the memory management seizes and the preloaded disposable is paused. Although this process takes some time, we do not worry much about it because it economizes memory on the long run, the biggest problems is that qmemman is synchronous, so only one request can be made at a time, anything that takes too much time on qmemman could prevent ballooning/balancing of other qubes on the system.
 
+Preloaded disposable's interruption
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-The actual passing of files between AppVM and a DisposableVM is implemented via qubes rpc.
+Preloaded disposables are paused for various reasons:
 
-Validating the DisposableVM savefile
-------------------------------------
+- Detection if the qube was used without being requested with :py:meth:`core-admin:qubes.vm.dispvm.DispVM.from_appvm`. Not every communication with a qube goes through :program:`qubesd`, it may go via Qrexec or GUI daemon;
+- CPU scheduling economy, domain is not eligible; and
+- Cronjob, timers and other things that don't block the init system and service manager completion won't run, they could possibly alter a clean state.
 
+But this comes at a cost:
 
-DisposableVM savefile contains references to template rootfs and to COW files. The COW files are restored before each DisposableVM start, so they cannot change. On the other hand, if templateVM is started, the template rootfs will change, and it may not be coherent with the COW files.
+- On early GUI connection, if done before the domain is paused:
```

Comment:
It's not clear the GUI is not connected before pausing. Maybe add a sentence about it, possibly linking to relevant issue if any?